scripts/start.sh

```diff
@@ -1,45 +1,45 @@
 #!/bin/bash
 
-STARTCOMMAND="./PalServer.sh"
+STARTCOMMAND=("./PalServer.sh")
 
 if [ -n "${PORT}" ]; then
-    STARTCOMMAND="${STARTCOMMAND} -port=${PORT}"
+    STARTCOMMAND+=("-port=${PORT}")
 fi
 
 if [ -n "${PLAYERS}" ]; then
-    STARTCOMMAND="${STARTCOMMAND} -players=${PLAYERS}"
+    STARTCOMMAND+=("-players=${PLAYERS}")
 fi
 
 if [ "${COMMUNITY}" = true ]; then
-    STARTCOMMAND="${STARTCOMMAND} EpicApp=PalServer"
+    STARTCOMMAND+=("EpicApp=PalServer")
 fi
 
 if [ -n "${PUBLIC_IP}" ]; then
-    STARTCOMMAND="${STARTCOMMAND} -publicip=${PUBLIC_IP}"
+    STARTCOMMAND+=("-publicip=${PUBLIC_IP}")
 fi
 
 if [ -n "${PUBLIC_PORT}" ]; then
-    STARTCOMMAND="${STARTCOMMAND} -publicport=${PUBLIC_PORT}"
+    STARTCOMMAND+=("-publicport=${PUBLIC_PORT}")
 fi
 
 if [ -n "${SERVER_NAME}" ]; then
-    STARTCOMMAND="${STARTCOMMAND} -servername=${SERVER_NAME}"
+    STARTCOMMAND+=("-servername=${SERVER_NAME}")
 fi
 
 if [ -n "${SERVER_PASSWORD}" ]; then
-    STARTCOMMAND="${STARTCOMMAND} -serverpassword=${SERVER_PASSWORD}"
+    STARTCOMMAND+=("-serverpassword=${SERVER_PASSWORD}")
 fi
 
 if [ -n "${ADMIN_PASSWORD}" ]; then
-    STARTCOMMAND="${STARTCOMMAND} -adminpassword=${ADMIN_PASSWORD}"
+    STARTCOMMAND+=("-adminpassword=${ADMIN_PASSWORD}")
 fi
 
 if [ -n "${QUERY_PORT}" ]; then
-    STARTCOMMAND="${STARTCOMMAND} -queryport=${QUERY_PORT}"
+    STARTCOMMAND+=("-queryport=${QUERY_PORT}")
 fi
 
 if [ "${MULTITHREADING}" = true ]; then
-    STARTCOMMAND="${STARTCOMMAND} -useperfthreads -NoAsyncLoadingThread -UseMultithreadForDS"
+    STARTCOMMAND+=("-useperfthreads" "-NoAsyncLoadingThread" "-UseMultithreadForDS")
```

Comment:
Don't all of these arguments still need spaces between them?

Reply:
Bash will place spaces between each element of the array, ensuring that your command is constructed with the necessary spaces between arguments. This is one of the benefits of using an array to construct commands - it handles the spacing for you, making the script more robust and less prone to errors related to incorrect spacing.